File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Mapper.Model/Fascicles/Implementation/FascicleModelMapper.cs
﻿using VecompSoftware.DocSuiteWeb.Entity.Commons;
using VecompSoftware.DocSuiteWeb.Entity.Dossiers;
using VecompSoftware.DocSuiteWeb.Entity.Fascicles;
using VecompSoftware.DocSuiteWeb.Entity.Processes;
using VecompSoftware.DocSuiteWeb.Model.Entities.Commons;
using VecompSoftware.DocSuiteWeb.Model.Entities.Dossiers;
using VecompSoftware.DocSuiteWeb.Model.Processes;
using VecompSoftwareFascicle = VecompSoftware.DocSuiteWeb.Model.Entities.Fascicles;

namespace VecompSoftware.DocSuiteWeb.Mapper.Model.Fascicles
{
    public class FascicleModelMapper : BaseModelMapper<Fascicle, VecompSoftwareFascicle.FascicleModel>, IFascicleModelMapper
    {
        #region [ Fields ]
        private readonly IMapperUnitOfWork _mapperUnitOfWork;
        #endregion

        #region [ Constructor ]
        public FascicleModelMapper(IMapperUnitOfWork mapperUnitOfWork)
        {
            _mapperUnitOfWork = mapperUnitOfWork;
        }
        #endregion

        #region [ Methods ]
        public override VecompSoftwareFascicle.FascicleModel Map(Fascicle entity, VecompSoftwareFascicle.FascicleModel modelTransformed)
        {
            modelTransformed.UniqueId = entity.UniqueId;
            modelTransformed.Year = entity.Year;
            modelTransformed.Number = entity.Number;
            modelTransformed.Conservation = entity.Conservation;
            modelTransformed.StartDate = entity.StartDate;
            modelTransformed.EndDate = entity.EndDate;
            modelTransformed.Name = entity.Name;
            modelTransformed.Title = entity.Title;
            modelTransformed.FascicleObject = entity.FascicleObject;
            modelTransformed.RegistrationDate = entity.RegistrationDate;
            modelTransformed.RegistrationUser = entity.RegistrationUser;
            modelTransformed.Manager = entity.Manager;
            modelTransformed.Rack = entity.Rack;
            modelTransformed.Note = entity.Note;
            modelTransformed.FascicleType = (VecompSoftwareFascicle.FascicleType)entity.FascicleType;
            modelTransformed.VisibilityType = (VecompSoftwareFascicle.VisibilityType)entity.VisibilityType;
            modelTransformed.MetadataValues = entity.MetadataValues;
            modelTransformed.MetadataDesigner = entity.MetadataDesigner;
            modelTransformed.DSWEnvironment = entity.DSWEnvironment;
            modelTransformed.CustomActions = entity.CustomActions;
            modelTransformed.ProcessLabel = entity.ProcessLabel;
            modelTransformed.DossierFolderLabel = entity.DossierFolderLabel;
            modelTransformed.FascicleDocumentUnits = _mapperUnitOfWork.Repository<IDomainMapper<FascicleDocumentUnit, VecompSoftwareFascicle.FascicleDocumentUnitModel>>().MapCollection(entity.FascicleDocumentUnits);
            modelTransformed.FascicleDocuments = _mapperUnitOfWork.Repository<IDomainMapper<FascicleDocument, VecompSoftwareFascicle.FascicleDocumentModel>>().MapCollection(entity.FascicleDocuments);
            modelTransformed.Category = entity.Category == null ? null : _mapperUnitOfWork.Repository<IDomainMapper<Category, CategoryModel>>().Map(entity.Category, new CategoryModel());
            modelTransformed.Container = entity.Container == null ? null : _mapperUnitOfWork.Repository<IDomainMapper<Container, ContainerModel>>().Map(entity.Container, new ContainerModel());
            modelTransformed.Contacts = entity.Contacts == null ? null : _mapperUnitOfWork.Repository<IDomainMapper<Contact, ContactModel>>().MapCollection(entity.Contacts);
            modelTransformed.DossierFolders = entity.DossierFolders == null ? null : _mapperUnitOfWork.Repository<IDomainMapper<DossierFolder, DossierFolderModel>>().MapCollection(entity.DossierFolders);
            modelTransformed.FascicleTemplate = entity.FascicleTemplate == null ? null : _mapperUnitOfWork.Repository<IDomainMapper<ProcessFascicleTemplate, ProcessFascicleTemplateModel>>().Map(entity.FascicleTemplate, new ProcessFascicleTemplateModel());
            modelTransformed.LastChangedDate = entity.LastChangedDate;
            return modelTransformed;
        }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Mappers/Fascicles/FascicleModelMapper.ts
import FascicleModel = require('App/Models/Fascicles/FascicleModel');
import BaseMapper = require('App/Mappers/BaseMapper');
import CategoryModelMapper = require('App/Mappers/Commons/CategoryModelMapper');
import ContactModelMapper = require('App/Mappers/Commons/ContactModelMapper');
import FascicleRoleModelMapper = require('App/Mappers/Fascicles/FascicleRoleModelMapper');
import FascicleDocumentModelMapper = require('App/Mappers/Fascicles/FascicleDocumentModelMapper');
import FascicleLinkModelMapper = require('App/Mappers/Fascicles/FascicleLinkModelMapper');
import FascicleDocumentUnitModelMapper = require('App/Mappers/Fascicles/FascicleDocumentUnitModelMapper');
import RequireJSHelper = require('App/Helpers/RequireJSHelper');
import ContainerModelMapper = require('App/Mappers/Commons/ContainerModelMapper');
import DossierFolderModelMapper = require('App/Mappers/Dossiers/DossierFolderModelMapper');
import TenantAOOModelMapper = require('App/Mappers/Tenants/TenantAOOModelMapper');
import MetadataRepositoryModelMapper = require('App/Mappers/Commons/MetadataRepositoryModelMapper');

class FascicleModelMapper extends BaseMapper<FascicleModel>{

    categoryMapper: typeof CategoryModelMapper;

    constructor() {
        super();
    }

    public Map(source: any): FascicleModel {

        const toMap: FascicleModel = {} as FascicleModel;

        if (!source) {
            return null;
        }

        const _fascicleDocumentUnitModelMapper: FascicleDocumentUnitModelMapper = RequireJSHelper.getModule<FascicleDocumentUnitModelMapper>(FascicleDocumentUnitModelMapper, 'App/Mappers/Fascicles/FascicleDocumentUnitModelMapper');
        const _fascicleDocumentModelMapper: FascicleDocumentModelMapper = RequireJSHelper.getModule<FascicleDocumentModelMapper>(FascicleDocumentModelMapper, 'App/Mappers/Fascicles/FascicleDocumentModelMapper');
        const _dossierFolderModelMapper: DossierFolderModelMapper = RequireJSHelper.getModule<DossierFolderModelMapper>(DossierFolderModelMapper, 'App/Mappers/Dossiers/DossierFolderModelMapper');
        const _tenantAOOModelMapper: TenantAOOModelMapper = RequireJSHelper.getModule<TenantAOOModelMapper>(TenantAOOModelMapper, 'App/Mappers/Tenants/TenantAOOModelMapper');
        const _metadataRepositoryModelMapper: MetadataRepositoryModelMapper = RequireJSHelper.getModule<MetadataRepositoryModelMapper>(MetadataRepositoryModelMapper, 'App/Mappers/Commons/MetadataRepositoryModelMapper');

        toMap.TenantAOO = source.TenantAOO ? _tenantAOOModelMapper.Map(source.TenantAOO) : null;
        toMap.Category = source.Category ? new CategoryModelMapper().Map(source.Category) : null;
        toMap.Container = source.Container ? new ContainerModelMapper().Map(source.Container) : null;
        toMap.Contacts = source.Contacts ? new ContactModelMapper().MapCollection(source.Contacts) : null;
        toMap.FascicleDocuments = source.FascicleDocuments ? _fascicleDocumentModelMapper.MapCollection(source.FascicleDocuments) : null;
        toMap.FascicleDocumentUnits = source.FascicleDocumentUnits ? _fascicleDocumentUnitModelMapper.MapCollection(source.FascicleDocumentUnits) : null;
        toMap.FascicleLinks = source.FascicleLinks ? new FascicleLinkModelMapper().MapCollection(source.FascicleLinks) : null;
        toMap.FascicleRoles = source.FascicleRoles ? new FascicleRoleModelMapper().MapCollection(source.FascicleRoles) : null;
        toMap.LinkedFascicles = source.LinkedFascicles ? new FascicleLinkModelMapper().MapCollection(source.LinkedFascicles) : null;
        toMap.DossierFolders = source.DossierFolders ? _dossierFolderModelMapper.MapCollection(source.DossierFolders) : [];
        toMap.MetadataRepository = source.MetadataRepository ? _metadataRepositoryModelMapper.Map(source.MetadataRepository) : null;

        toMap.FascicleObject = source.FascicleObject;
        toMap.FascicleType = source.FascicleType;
        toMap.Manager = source.Manager;
        toMap.Name = source.Name;
        toMap.Note = source.Note;
        toMap.Number = source.Number;
        toMap.Rack = source.Rack;
        toMap.RegistrationDate = source.RegistrationDate;
        toMap.RegistrationUser = source.RegistrationUser;
        toMap.LastChangedDate = source.LastChangedDate;
        toMap.LastChangedUser = source.LastChangedUser;
        toMap.StartDate = source.StartDate;
        toMap.Title = source.Title;
        toMap.UniqueId = source.UniqueId;
        toMap.VisibilityType = source.VisibilityType;
        toMap.Year = source.Year;
        toMap.MetadataValues = source.MetadataValues;
        toMap.MetadataDesigner = source.MetadataDesigner;
        toMap.CustomActions = source.CustomActions;
        toMap.ProcessLabel = source.ProcessLabel;
        toMap.DossierFolderLabel = source.DossierFolderLabel;
        toMap.EndDate = source.EndDate;
        toMap.Conservation = source.Conservation;

        return toMap;
    }
}

export =FascicleModelMapper;